packages/babel-plugin-react-jsx/src/TransformJSXToReactBabelPlugin.js

```diff
@@ -576,25 +614,108 @@ module.exports = function(babel) {
     },
   });
 
+  const createIdentifierName = (path, autoImport, name, importName) => {
+    if (autoImport === IMPORT_TYPES.none) {
+      return `React.${name}`;
+    } else if (autoImport === IMPORT_TYPES.namedExports) {
+      const identifierName = `${importName[name]}`;
+      return identifierName;
+    } else {
+      return `${importName}.${name}`;
+    }
+  };
+
+  function addAutoImports(path, state) {
+    if (state.autoImport === IMPORT_TYPES.none) {
+      return;
+    }
+
+    if (IMPORT_TYPES[state.autoImport] === undefined) {
+      throw path.buildCodeFrameError(
+        'autoImport must be one of the following: ' +
+          Object.keys(IMPORT_TYPES).join(', '),
+      );
+    }
+    if (state.useCreateElement) {
+      throw path.buildCodeFrameError(
+        'auto importing cannot be used with createElement. Consider setting ' +
```

Comment:
```suggestion
        'autoImport cannot be used with `createElement()`. Consider setting ' +
```
Alternately, "Automatically importing..."?